src/stk500.c

```diff
@@ -116,7 +116,7 @@ int stk500_getsync(const PROGRAMMER *pgm) {
 
   for (attempt = 0; attempt < max_sync_attempts; attempt++) {
     // Restart Arduino bootloader for every sync attempt
-    if (str_eq(pgm->type, "Arduino") && attempt > 0) {
+    if (PDATA(pgm)->autoreset && attempt > 0) {
```

Comment:
I think at this place the check `str_eq(pgm->type, "Arduino")` must remain. It's OK to have the *additional* constraint `PDATA(pgm)->autoreset` that the user controls. Only when the `-c` programmer is of the Arduino type (ie, if it's `-c arduino`) should the driver trigger reset here. I believe, this has to do with the weird way optiboot has at some point decided when to jump to the application and when to jump to the bootloader. Normal STK500 programmers don't need this and it would prossibly be harmful to the re-sync attempts.

Reply:
Yes, but we only provide auto-reset functionality for the `-c arduino programmer`. So this should not run when using just another stk500v1 based programmer

Reply:
That is a good point. But just imagine someone *adds* the autoreset functionality for *another* stk500.c based programmer (just like arduino did that). Then this code would run for that programmer, too. But I don't think this will be needed. I think it's *only* optiboot that needs that code at this point. `str_eq(pgm->type, "Arduino")` is kinda a proxy for optiboot (the best AVRDUDE can do outside what urclock has been doing by examining the bootloader itself and computing hash sums).

TLDR; I would use `if(str_eq(pgm->type, "Arduino") && PDATA(pgm)->autoreset && attempt > 0)` to document it's intended for `-c arduino` even though this might be redundant right now.

Reply:
Good point! Thanks for the explaination

Reply:
Actually, thinking about this more: as autoreset is introduced in the stk500.c PDATA and (one way or another) defaults to true this code *will* run for *all* stk500 programmers changing behaviour. It defo needs to be conditioned on Arduino.